qutebrowser/keyinput/modeparsers.py

```diff
@@ -230,3 +230,47 @@ def __init__(self, win_id, parent=None):
         super().__init__(win_id, parent, supports_count=True,
                          supports_chains=True)
         self.read_config('caret')
+
+
+class MarkKeyParser(keyparser.CommandKeyParser):
+
+    """KeyParser for set_mark and jump_mark mode.
+
+       Attributes:
+           _mode: Either KeyMode.set_mark or KeyMode.jump_mark.
+    """
+
+    def __init__(self, win_id, mode, parent=None):
+        super().__init__(win_id, parent, supports_count=False,
+                         supports_chains=False)
+        self._mode = mode
+
+    def handle(self, e):
+        """Override handle to always match the next key and create a mark.
+
+        Args:
+            e: the KeyPressEvent from Qt.
+
+        Return:
+            True if event has been handled, False otherwise.
+        """
+
+        if utils.keyevent_to_string(e) is None:
+            # this is a modifier key, let it pass and keep going
+            return True
+
+        if not e.text().isalpha() and e.text() != "'":
+            # only valid mark names are [a-zA-Z']
+            message.error(self._win_id, e.text() + " isn't a valid mark")
+            return True
+
+        if self._mode == usertypes.KeyMode.set_mark:
+            self._commandrunner.run('set-mark "{}"'.format(e.text()))
+        elif self._mode == usertypes.KeyMode.jump_mark:
+            self._commandrunner.run('jump-mark "{}"'.format(e.text()))
```

Comment:
Ah, at least with `jump_mark` that's not easily possible as it's in `browser/commands.py`. I'll need to think about it some more, it might be okay as-is.